force-app/main/default/classes/SOQL.cls

```diff
@@ -85,9 +103,17 @@ public with sharing class SOQL {
      * @return A map of Task Ids to Task records.
      * Update the corresponding test class and testGetTasksDueNextWeek method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
+    public static Map<Id,Task> getTasksDueNextWeek () {
     //  The body of the method
-    //}
+        List<Task> allTasksDueIn7Days = [SELECT Id, WhatId, Subject, ActivityDate, Status, Priority, OwnerId, AccountId, isClosed FROM Task where ActivityDate > TODAY and  ActivityDate <= NEXT_N_DAYS:7];
+        Map<Id, Task> tasksMap = new Map<Id, Task>();
+
+        for (Task taskRec : allTasksDueIn7Days) {
+            tasksMap.put(taskRec.Id, taskRec);
+        }
+
```

Comment:
There is a special constructor for Map<Id, SObject> that takes a List and separates it into a map of the Id to the record itself. You can use this to return the keySet() of a map, which is a bit simpler (and somehow faster) than doing the iteration on lines 109 through 113.